modules/apps/portal-crypto/portal-crypto-hash-api/src/main/java/com/liferay/portal/crypto/hash/response/HashResponse.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.crypto.hash.response;
+
+import java.util.Optional;
+
+/**
+ * @author Carlos Sierra Andrés
+ * @author Arthur Chan
+ */
+public class HashResponse {
+
+	public byte[] getHash() {
+		return _hash;
+	}
+
+	public Optional<byte[]> getPepper() {
+		return _pepper;
+	}
+
+	public Optional<byte[]> getSalt() {
+		return _salt;
+	}
+
+	public static class Builder {
```

Comment:
I don't think we need a builder for the `HashResponse` since it is never going to be built by the user of the API, is it?
we can just return the impl directly with the values.

Reply:
Will do